app/lib/search/index_simple.dart

```diff
@@ -231,7 +237,28 @@ class SimplePackageIndex implements PackageIndex {
         return Score.max([name, descr, readme]).removeLowValues(
             fraction: 0.01, minValue: 0.001);
       }).toList();
-      return Score.multiply(wordScores);
+      Score score = Score.multiply(wordScores);
+
+      // filter results based on exact phrases
+      final List<String> exactPhrases = extractExactPhrases(text);
+      if (exactPhrases.isNotEmpty) {
+        final Map<String, double> matched = <String, double>{};
+        for (String package in score.getKeys()) {
+          final allText = _normalizedPackageText[package];
+          bool matchedAllPhrases = true;
+          for (String phrase in exactPhrases) {
+            final String normalizedPhrase = normalizeBeforeIndexing(phrase);
+            final bool matchedPhrase = allText.contains(normalizedPhrase);
+            matchedAllPhrases = matchedAllPhrases && matchedPhrase;
```

Comment:
We can actually break out of the loop as soon as `matchedPhrase == false`.
e.g.

```dart
next_package:
for (String package in score.getKeys()) {
  final allText = _normalizedPackageText[package];
  for (String phrase in exactPhrases) {
    if (!allText.contains(phrase)) {
      continue next_package;
    }
    matched[package] = score[package];
  }
}

```